### PR TITLE
Ensure train/n_updates metric accounts for early stopping of training loop

### LINKED ISSUE #1229: [Bug]: PPO - n_updates is inaccurate when using target_kl and confusing breaking variable
### 🐛 Bug

In `stable_baselines3/ppo/ppo.py`, passing `target_kl` parameter causes early stopping to occur during the update loop. This is not reflected in the logged `train/n_updates` metric. 
Changing line 278 from `self._n_updates += self.n_epochs` to `self._n_updates += epoch` would fix it.

There also is an extra `continue_training` parameter used with this logic that is not needed. It already breaks in the `self.target_kl` check area of line 262. Removing `continue_training` and its usage shouldn't affect current functionality. But I'm unsure if that is the intended functionality. 

Due to the break statement in the `self.target_kl` check, it is possible to not update due to high `approx_kl_div`. Was the intended break supposed to be after the optimizer step (using `continue_training` ) or before (breaking after checking `target_kl`)? 

### To Reproduce

N/A

### Relevant log output / Error message

```shell
N/A
```

### System Info

N/A

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

### PR DESCRIPTION
**Pull request by adamfrly**
## Description

This PR closes #1229. A new variable `num_updates` is created before the training loop and is initially set to `self.n_epochs`. If the loop does not stop early, everything is effectively the same. However, if the loop is stopped early due to `kl_target` being set, `num_updates` is set to the number of epochs that have occurred plus the current epoch (i.e. `num_updates = epoch + 1`).

This should ensure that `train/n_updates` is always the correct value.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Review by araffin — COMMENTED** [review]
@adamfrly I've simplified the change ;), could you confirm is does solve the issue?

**Comment by adamfrly** [comment]
@araffin Hahaha thanks for the change, looks much cleaner :)

And yes, it does solve the issue.

**Review by araffin — APPROVED** [review]
